File: src/Scripting/PythonPY/adapter.py
class LegacyFahrenheitSensor:
    def read_fahrenheit(self) -> int:
        return 86


class FahrenheitSensorAdapter:
    def __init__(self, sensor: LegacyFahrenheitSensor) -> None:
        self._sensor = sensor

    def read_celsius(self) -> int:
        return (self._sensor.read_fahrenheit() - 32) * 5 // 9


def main() -> None:
    legacy = LegacyFahrenheitSensor()
    reader = FahrenheitSensorAdapter(legacy)
    print(f"legacy={legacy.read_fahrenheit()}F")
    print(f"adapted={reader.read_celsius()}C")


if __name__ == "__main__":
    main()

File: src/Web/TypeScriptTS/adapter.ts
interface TemperatureReader {
  readCelsius(): number;
}

class LegacyFahrenheitSensor {
  readFahrenheit(): number {
    return 86;
  }
}

class FahrenheitSensorAdapter implements TemperatureReader {
  constructor(private readonly sensor: LegacyFahrenheitSensor) {}

  readCelsius(): number {
    return Math.trunc(((this.sensor.readFahrenheit() - 32) * 5) / 9);
  }
}

const legacy = new LegacyFahrenheitSensor();
const reader: TemperatureReader = new FahrenheitSensorAdapter(legacy);
console.log(`legacy=${legacy.readFahrenheit()}F`);
console.log(`adapted=${reader.readCelsius()}C`);
